src/completion/widget.ts

```diff
@@ -149,20 +149,24 @@ class CompletionWidget extends Widget {
    * Handle `after_attach` messages for the widget.
    *
    * #### Notes
-   * Captures document events in capture phase to dismiss or navigate the
+   * Captures window events in capture phase to dismiss or navigate the
    * completion widget.
+   *
+   * Because its parent (reference) widgets use window listeners instead of
+   * document listeners, the completion widget must also use window listeners
+   * in the capture phase.
```

Comment:
@blink1073 Oops, I forgot that I'm the one who originally said let's switch to `document` ... sorry! But yeah, the solution still seems to be this.